src/main/scala/za/co/absa/commons/BuildInfo.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.commons
+
+import za.co.absa.commons.lang.ARM.using
+import za.co.absa.commons.lang.ImmutableProperties
+
+/**
+  * Returns the project build version taken from the build.properties file.
+  * It's particularly convenient for Maven-based project. Just copy 'build.properties.template'
+  * file to your classpath (without 'template'-suffix) and enable Maven resource filtering.
+  */
+object BuildInfo {
+  val BuildProps: ImmutableProperties =
+    using(this.getClass.getResourceAsStream("/build.properties"))(ImmutableProperties.fromStream)
```

Comment:
Will fix NPE, thx.
Regarding configurable file name, can you suggest an easy approach? The point is that `BuildInfo` is a singleton, so no initialization is needed. You just use it. If you want to have the filename parametrized then it should be done via another static value or even JVM arg **before the first usage** of the `BuildInfo` object.

Reply:
I think it doesn't make sense to parametrize this with another value just for this small file. The easiest way I could think of is to pass it as a parameter to `BuildInfo`.

Reply:
Ok, I'll add `apply` method to allow for this type of usage, just in case anybody needs it. Sounds simple enough.

Reply:
What about this?
```scala
// define your own BuildInfo singleton
object MyBuildInfo extends BuildInfo("my") // will lookup my.properties
// ... and then use throughout the code
MyBuildInfo.Version
```

Reply:
Property names could also be optionally parametrized:
```scala
object MyBuildInfo extends BuildInfo(
  resourceName = "my",
  versionKey = "ver",
  timestampKey = "time"
)
```

Reply:
Yes, I think this gives maximum flexibility if needed and still keeps maximum simplicity for the standard case

Reply:
done.
I've also added `apply()` method with the default args for even more code style freedom. Hope it will be useful at least for someone at some time.
```scala
val myVersion = BuildInfo(resourcePrefix = "/example/my").Version 
```

I've also moved it to the `buildinfo` package as one more class with it's companion was added.